### PR TITLE
[14.0][MIG] stock_picking_group_by_partner_by_carrier_sale_line_position

### PR DESCRIPTION
**Pull request by hailangvn**
Depends on:
* `sale_order_line_position`: https://github.com/OCA/sale-reporting/pull/118
* `delivery_line_sale_line_position`: https://github.com/OCA/stock-logistics-reporting/pull/164
* https://github.com/OCA/stock-logistics-workflow/pull/908

### CODE REVIEW

**Review by mmequignon — APPROVED** [review]

File: stock_picking_group_by_partner_by_carrier_sale_line_position/models/stock_picking.py [hunk 1]
```diff
@@ -1,18 +0,0 @@
...
-
-    def _get_sorted_moves(self):
-        self.ensure_one()
-        return self.move_lines.sorted(
-            lambda m: m.sale_line_id.order_id.id * 1000 + m.sale_line_id.position
-        )
-
-    def _get_sorted_move_lines(self):
-        self.ensure_one()
-        return self.move_line_ids.sorted(lambda l: l.move_id.sale_line_id.position)
```
Comment (sebalix) [inline stock_picking_group_by_partner_by_carrier_sale_line_position/models/stock_picking.py]: I do not understand this change (I don't know this module very well too), could you explain why we don't need to sort anymore?

Comment (hailangvn) [reply stock_picking_group_by_partner_by_carrier_sale_line_position/models/stock_picking.py]: Good day @sebalix, I could not find any place that call those methods.

Good day @TDu, Could you please help to advise the purpose of the methods when adding with
https://github.com/OCA/stock-logistics-reporting/pull/155/files#diff-5e837d58a39e022c613d0595b6d430527a060110365d3e2a1e5e76e612553d93R1-R18?

https://github.com/search?q=org%3AOCA+_get_sorted_moves&type=code
https://github.com/search?q=org%3AOCA+_get_sorted_move_lines&type=code
https://github.com/odoo/odoo/search?q=_get_sorted_moves&type=commits
https://github.com/odoo/odoo/search?q=_get_sorted_move_lines&type=commits

Comment (hailangvn) [reply stock_picking_group_by_partner_by_carrier_sale_line_position/models/stock_picking.py]: Thanks @sebalix. I reverted the change.

File: stock_picking_group_by_partner_by_carrier_sale_line_position/report/report_deliveryslip.xml [hunk 2]
```diff
@@ -27,19 +27,6 @@
...
-    <!-- Add sale line position on remaining to deliver list -->
-    <xpath expr="//tr[@t-foreach='remainings']/td[1]/span[1]" position="before">
-      <span
-                t-if="remaining.get('sale_order_line')"
-                t-esc="remaining['sale_order_line'].position_formatted"
-            />
-    </xpath>
-
-  </template>
+  <!-- Deliver list does not have remainings any more. -->
```
Comment (sebalix) [inline stock_picking_group_by_partner_by_carrier_sale_line_position/report/report_deliveryslip.xml]: This shouldn't be removed, some improvements done on `13.0` haven't been ported to `14.0`, I started to port all missing fix/features on the base module there:
- https://github.com/OCA/stock-logistics-workflow/pull/908

I block the current one until the PR linked is fixed/merged.

**Review by sebalix — CHANGES_REQUESTED** [review]

File: stock_picking_group_by_partner_by_carrier_sale_line_position/report/report_deliveryslip.xml (hunk 2 above)
Comment (sebalix) [reply stock_picking_group_by_partner_by_carrier_sale_line_position/report/report_deliveryslip.xml]: @hailangvn the PR mentionned above has been merged and thus `stock_picking_group_by_partner_by_carrier` is now up-to-date, you can rebase your PR on top on 14.0

Comment (sebalix) [reply stock_picking_group_by_partner_by_carrier_sale_line_position/report/report_deliveryslip.xml]: The `remaining` stuff is now present in the report.

**Comment by hailangvn** [comment]
Thanks @sebalix for the work. This PR has been updated

**Comment by simahawk** [comment]
@sebalix can you update your review?

**Review by TDu — APPROVED** [review]

**Review by sebalix — APPROVED** [review]

**Comment by sebalix** [comment]
@hailangvn thanks for the update!

**Pull request merged**